### ISSUE TITLE
[REQUEST] Implement route type verification (GTFS rule)

### ISSUE THREAD

**Issue by maximearmstrong** [MobilityData/gtfs-validator#145]
**Is your feature request related to a problem? Please describe.**
Verifies if the route type is valid. This is a GTFS rule implemented in Google Python validator and featured in Google Type Error as TYPE_INVALID_ROUTE_TYPE.

**Describe the solution you'd like**
Actual Google GTFS validator behaviour : verifies if route_type is valid (int included in the _ROUTE_TYPE_IDS).

**Describe alternatives you've considered**

**Additional context**
Line 68 in Error support priorities https://docs.google.com/spreadsheets/d/1vqe6wq7ctqk1EhYkgtZ0_TbcQ91vccfs2daSjn20BLE/edit#gid=0

**Comment by barbeau** [comment]
For this rule, note that official GTFS route_types are here:
https://gtfs.org/reference/static#routestxt

...although Google has a history of supporting a number of extended route types that aren't in the official spec:
https://developers.google.com/transit/gtfs/reference/extended-route-types

So this is another rule where we will likely need to support configuration for route types that Google supports, vs. the official route_types in the GTFS spec.

**Comment by fabrice-v** [comment]
Oh yeah. I think that corresponds to this part of the GTFS schema we have in the resources
https://github.com/MobilityData/gtfs-validator/blob/a2b96ae0783d770fd2a1fa3da7c8cacd0461a473/adapter/repository/in-memory-simple/src/main/resources/gtfs_spec.asciipb#L2003
But, we don't presently use the `namedvalue` mechanism in the protobuf of which I forgot the details. 
I think we will work on providing those configurations through `execution-parameters.json`

**Comment by barbeau** [comment]
FWIW, it looks like the Python validator only supports 0-7 as valid route types - from https://github.com/google/transitfeed/blob/master/transitfeed/route.py#L32:

```python
  _ROUTE_TYPES = {
    0: {'name':'Tram', 'max_speed':100},
    1: {'name':'Subway', 'max_speed':150},
    2: {'name':'Rail', 'max_speed':300},
    3: {'name':'Bus', 'max_speed':100},
    4: {'name':'Ferry', 'max_speed':80},
    5: {'name':'Cable Car', 'max_speed':50},
    6: {'name':'Gondola', 'max_speed':50},
    7: {'name':'Funicular', 'max_speed':50},
    }
```

**Comment by maximearmstrong** [comment]
Rule implemented in PR #217
